code/go/internal/yamlschema/schema_spec.go

```diff
@@ -59,28 +59,18 @@ func (i *itemSchemaSpec) resolve(target semver.Version) (map[string]interface{},
 }
 
 func (i *itemSchemaSpec) patchForVersion(target semver.Version) ([]byte, error) {
-	var patch []byte
+	var patch []any
 	for _, version := range i.Versions {
 		if sv, err := semver.NewVersion(version.Before); err != nil {
 			return nil, err
 		} else if !target.LessThan(sv) {
 			continue
 		}
 
-		patchData, err := json.Marshal(version.Patch)
-		if err != nil {
-			return nil, fmt.Errorf("failed to marshal patch: %w", err)
-		}
-
-		if len(patch) == 0 {
-			patch = patchData
-			continue
-		}
-
-		patch, err = jsonpatch.MergeMergePatches(patch, patchData)
```

Comment:
This doesn't do what I thought. The library supports JSON patches, and "merge patches", this method is used to merge merge patches, but here we are working with JSON patches. JSON patches can be merged just by concatenating their actions.